### PR TITLE
Update rule about "when" indentation

### PR DESCRIPTION
**Pull request by whatyouhide**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,13 +64,18 @@
...
   ```elixir
   def format_error({exception, stacktrace})
       when is_list(stacktrace) and stacktrace != [] do
-    #...
+    # ...
+  end
+
+  defp format_exit({exception, stacktrace} = reason, joiner)
+       when is_list(stacktrace) and stacktrace != [] and is_binary(joiner) do
+    # ...
```
Comment (eksperimental) [inline README.md:78]: defmacrop example would be good

Comment (whatyouhide) [reply README.md:78]: It's the same effect @eksperimental, I think it's fine like this as well :) Also `defmacrop` is not used very commonly IMO.

Comment (b0oh) [reply README.md:78]: I think @eksperimental point was if it is 4 spaces or aligned to a `defp`, `defmacro` would show clear that it is 4 spaces, not aligned to a previous line.

Comment (eksperimental) [reply README.md:78]: @meatinleather exactly, also if there was a really long function that is rarely used, I still would choose that one to show that no matter how long it is, it has to be aligned to the identifier

Comment (whatyouhide) [reply README.md:78]: @meatinleather it _is_ aligned to the previous line, it's not always four spaces.

``` elixir
defmacro foo(...)
         when ... do
  ...
end
```

Comment (b0oh) [reply README.md:78]: :O, okey then, then `defmacro` definitely would be useful.

Comment (eksperimental) [reply README.md:78]: defmacro give an identation of 9, so it's good as an example, IMO

**Pull request merged**